src/AutoMapper/QueryableExtensions/ProjectionMappers/EnumerableProjectionMapper.cs

```diff
@@ -34,8 +34,18 @@ public Expression Project(IGlobalConfiguration configuration, MemberMap memberMa
             }
             if (!memberMap.DestinationType.IsAssignableFrom(sourceExpression.Type))
             {
-                var convertFunction = memberMap.DestinationType.IsArray ? ToArrayMethod : ToListMethod;
-                sourceExpression = Call(convertFunction.MakeGenericMethod(destinationListType), sourceExpression);
```

Comment:
```C#
converFunction = convertFunction.MakeGenericMethod(destinationListType);
if(memberMap.DestinationType.IsAssignableFrom(convertFuncion.ReturnType))
{
    sourceExpression = Call(convertFunction.MakeGenericMethod(destinationListType), sourceExpression);
}
else
{
     // new code
}
```
Let's revert the caching in the `PropertyMap` and have everything here. Even if it were useful, which I don't think iit is, this is a corner case.

Reply:
Sure, updated.